integration_tests/tests/integration/simulation/run.rs

```diff
@@ -0,0 +1,662 @@
+                &self.task,
+                &self.state.vdaf,
+                report_time,
+                &self.components.http_client,
+            )
+            .await
+            {
+                warn!(?error, "client error");
+                // We expect to receive an error if the report timestamp is too far away from the
+                // current time, so we'll allow errors for now.
+            }
+        }
+        ControlFlow::Continue(())
+    }
+
+    async fn execute_garbage_collector(&mut self, role: AggregatorRole) -> ControlFlow<TestResult> {
+        let garbage_collector = match role {
+            AggregatorRole::Leader => &self.components.leader_garbage_collector,
+            AggregatorRole::Helper => &self.components.helper_garbage_collector,
+        };
+        if let Err(error) = garbage_collector.run().await {
+            error!(?error, "garbage collector error");
+            return ControlFlow::Break(TestResult::error(format!("{error:?}")));
+        }
+        ControlFlow::Continue(())
+    }
+
+    async fn execute_key_rotator(&mut self, role: AggregatorRole) -> ControlFlow<TestResult> {
+        let key_rotator = match role {
+            AggregatorRole::Leader => &self.components.leader_key_rotator,
+            AggregatorRole::Helper => &self.components.helper_key_rotator,
+        };
+        if let Err(error) = key_rotator.run().await {
+            error!(?error, "key rotator error");
+            return ControlFlow::Break(TestResult::error(format!("{error:?}")));
+        }
+        ControlFlow::Continue(())
+    }
+
+    async fn execute_aggregation_job_creator(&mut self) -> ControlFlow<TestResult> {
+        let aggregation_job_creator = Arc::clone(&self.components.aggregation_job_creator);
+        let task = Arc::clone(&self.leader_task);
+        if let Err(error) = aggregation_job_creator
+            .create_aggregation_jobs_for_task(task)
+            .await
+        {
+            error!(?error, "aggregation job creator error");
+            return ControlFlow::Break(TestResult::error(format!("{error:?}")));
+        }
+        ControlFlow::Continue(())
+    }
+
+    async fn execute_aggregation_job_driver(&mut self) -> ControlFlow<TestResult> {
+        let leases = match (self.components.aggregation_job_driver_acquirer_cb)(10).await {
+            Ok(leases) => leases,
+            Err(error) => {
+                error!(?error, "aggregation job driver error");
+                return ControlFlow::Break(TestResult::error(format!("{error:?}")));
+            }
+        };
+        debug!(count = leases.len(), "acquired aggregation jobs");
+        for lease in leases {
+            if let Err(error) = (self.components.aggregation_job_driver_stepper_cb)(lease).await {
+                if let aggregator::Error::Http(_) = error {
+                    warn!(?error, "aggregation job driver error");
+                    return ControlFlow::Continue(());
+                }
```

Comment:
Hmm? Is this to support reusing this code for the fault injection case?

Not sure how I feel about it--if there's a legitimate bug that triggers an `::Http` error, then it won't be caught, but maybe such a bug is better caught by unit tests.